frontend/src/pages/v2/StaticCanvas.tsx

```diff
@@ -39,14 +39,21 @@ const StaticCanvas = ({ elements, layers, onLayersUpdate }: StaticCanvasProps) =
     reactFlowInstance.fitView();
   };
 
-  const doubleClickNode = (node: Node) => {
-    if (node.data['taskType'] !== TaskType.DAG) {
-      return;
-    }
-    const newLayers = [...layers, getTaskKeyFromNodeKey(node.id)]; //remove `task.`
+  const subDagExpand = (nodeKey: string) => {
+    const newLayers = [...layers, getTaskKeyFromNodeKey(nodeKey)]; //remove `task.`
     onLayersUpdate(newLayers);
   };
 
+  elements.forEach(elem => {
+    // For each SubDag node, provide a callback function if expand button is clicked.
+    if (elem && elem.type === NodeTypeNames.SUB_DAG && elem.data) {
+      elem.data = {
+        label: elem.data.label,
+        expand: subDagExpand,
+      } as SubDagFlowElementData;
```

Comment:
Why do we need to cast type?

Reply:
Because the base element for `data` is `FlowElementDataBase`, and I want to claim that this var is a child of `FlowElementDataBase`, which is `SubDagFlowElementData`, so cast the type to this child interface.

Reply:
@zijianjoy understood, I don't think the current code is what you want to achieve.

Note, if you make a typing mistake in the object, type checker will ignore the error because this is a type cast.

To have proper type checking, you can write:
```
const data: SubType = ...
elem.data = data
```
In this way, type is checked at both statements without casting.